src/Microsoft.Sbom.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -64,28 +64,11 @@ public static IServiceCollection AddSbomTool(this IServiceCollection services, L
     {
         services
             .AddSingleton<IConfiguration, Configuration>()
-            .AddTransient(_ => FileSystemUtilsProvider.CreateInstance())
+            .AddTransient(_ => FileSystemUtilsProvider.CreateInstance(CreateLogger(logLevel)))
             .AddTransient(x =>
             {
                 logLevel = x.GetService<InputConfiguration>()?.Verbosity?.Value ?? logLevel;
```

Comment:
Do we need to use this line above to determine the log level of the FileSystemUtilsProvider logger too? It seems like the way this is currently written that new logger would be ignoring any user inputted log level

Reply:
My first iteration was exactly what you describe, since I wanted to log the Exception at `Debug` level as suggested in the previously existing TODO comment. Unfortunately, those attempts failed because the lambda was being called on a different `IServiceProvider` than the one that actually contains the user config. This disconnect is why I ended up logging the Exception at `Warning` level, so it would always show up if it occurred.

Reply:
Makes sense!